### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: src/Gir/Generation/IGeneratableExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.IO;
+
+namespace Gir
+{
+	public static class IGeneratableExtensions
+	{
+		public static IndentWriter GetWriter (this IGeneratable gen, GenerationOptions opts)
+		{
+			var path = Path.Combine(opts.DirectoryPath, gen.Name + ".cs");
```
Comment (stsundermann) [inline src/Gir/Generation/IGeneratableExtensions.cs]: ".cs" can be a constant

Comment (Therzok) [reply src/Gir/Generation/IGeneratableExtensions.cs]: Done.

File: src/Gir/Marshal/IType.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using System;
+namespace Gir
+{
+	public interface IType
```
Comment (stsundermann) [inline src/Gir/Marshal/IType.cs]: Is this complete yet? Is IType supposed to give information only about the native type or do we also add information on C# like CSharpType or IsBlittable here?

Comment (Therzok) [reply src/Gir/Marshal/IType.cs]: We can make the blittable and co extensions. I'm trying to keep the interface simple, as in, what's the minimum set of things a marshallable type needs to have. It's not complete yet.

**Pull request merged**